### DISCUSSION TITLE
Is the bounds option working?

### DISCUSSION THREAD

**Discussion by germi** [vue-leaflet/vue-leaflet#176] [General]
It doesn't seem to be. I change the bounds prop but it doesn't do anything.
Am I missing something, or this has not been implemented yet?

Thanks!

**Comment by mikeu** [comment]
Hi @germi, thanks for the question! As far as I'm aware, the initial value of the `:bounds` property is not used on map creation simply because Leaflet itself does not accept a bounds option on new `L.map` instances. However, changing the value of that prop later on is causing the map to shift its display to the new bounds as expected, in a quick local test for me. Are you able to share a sample of the code and bounds values you're tried?

Depending on what exactly you're expecting, the one possible "gotcha" I can think of is that we are using Leaflet's `fitBounds` method on the map to ensure that the new bounds are entirely visible within the display -- but _not_ stretching and skewing the map or view so that the specified rectangle fits _exactly_ in the component. It still sticks to the allowed zoom levels and the aspect ratio of the map div container. So it is possible that relatively minor changes to the bounds simply won't cause an update to the displayed view.

If that is the issue, you could try supplying a smaller value for the [`zoomSnap` option](https://leafletjs.com/reference.html#map-zoomsnap) (default is 1, so e.g. `:options="{ zoomSnap: 0.1 }"), to see if smaller changes start visibly altering the map extent closer to how you're expecting.

**Comment by germi** [comment]
Hi,

Sorry for the late reply (I just noticed this...) and thanks for your swift answer.
Unfortunately it's still not working for me...

I have created this wrapper component for my map. It accepts an array of `markers` as a prop
I'm computing the `bounds` of the map from the markers, as well as the `displayCenter` computed prop.

When I change the array of markers, the center of the map gets repositioned correctly, but the map does not "fit to bounds".

Could it be a combination of the center + zoom options that is not "letting" the map show the necessary bounds to display all the markers?

Thanks a lot

```
<template>
  <l-map
    ref="map"
    :maxZoom="16"
    :minZoom="7"
    :zoom="zoom"
    :center="displayCenter"
    :bounds="bounds"
    :options="{ zoomSnap: 0.1 }"
    @move="move"
    @ready="onReady"
    @update:bounds="boundsUpdated"
  >
    <l-tile-layer :url="url"></l-tile-layer>

    <l-marker
      v-for="marker in markers"
      :key="marker.id"
      :id="marker.id"
      :options="{ id: marker.id }"
      :lat-lng="marker.geo"
      :class="marker.id"
      @click="markerClick"
    >
    </l-marker>
  </l-map>
</template>

<script>
// https://github.com/vue-leaflet/vue-leaflet
// https://github.com/vue-leaflet/vue3-demo-project/blob/master/src/App.vue
import "leaflet/dist/leaflet.css";
import { LMap, LTileLayer, LMarker, LIcon } from "@vue-leaflet/vue-leaflet";
import { latLngBounds } from "leaflet";

import { extent } from "d3";

export default {
  components: {
    LMap,
    LTileLayer,
    LMarker,
    LIcon,
  },
  props: {
    markers: {
      type: Array,
      default: () => [],
    },
  },
  data() {
    return {
      url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
      // url: "https://stamen-tiles-{s}.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}{r}.png",
      zoom: 13,
      center: [41.390205, 2.154007],
      iconWidth: 25,
      iconHeight: 40,
      boundsFromMap: null,
    };
  },
  computed: {
    displayCenter() {
      if (this.markers.length) {
        const boundsX = extent(this.markers, (d) => d.geo[0]);
        const boundsY = extent(this.markers, (d) => d.geo[1]);

        return [(boundsX[0] + boundsX[1]) / 2, (boundsY[0] + boundsY[1]) / 2];
      }
      return this.center;
    },
    bounds() {
      return latLngBounds(this.markers.map((e) => e.geo));
    },
  },
  methods: {
    move(e) {
      // console.log(e);
    },
    markerClick(e) {
      this.$emit("markerClick", e.target.options.id);
    },
    boundsUpdated(bounds) {
      console.log("updated bounds", bounds);
      console.log("bounds that should be", this.bounds);
      this.boundsFromMap = bounds;
    },
  },
};
</script>
```

**Comment by germi** [comment]
Basically it looks like the zoom does not "update" when the bounds change, that's why all the markers can' be seen

**Comment by germi** [comment]
Actually I'm also trying directly this: https://vue2-leaflet.netlify.app/examples/set-bounds.html

and it doesn't seem to work like in the demo site.

I know it's for Vue2.. I just changed the import to be the "@vue-leaflet/vue-leaflet", but it doesn't work (I assumed all the API would be the same, all the props, etc)

**Comment by germi** [comment]
ok I think I could figure it out...

the bounds prop cannot be the object returned from latLngBounds(), which is in the shape:

```js
{
_northEast: {lat: ... , lng: ...},
_southWest: {lat: ... , lng: ...}
}
```

it has to be transformed into the array version:

```js
[[lat, lng], [lat, lng]]
```
this seems to do the trick...

Also, in LeaFlet I see the option to set a `padding` for the FitBounds option (passed as second param to the `fitBounds()` function. Is there a way to pass this as a prop in the `<l-map>` component?

**Reply by mikeu** [reply]
Hi @germi , now it's my turn to be sorry for a _very_ late reply.

At first glance, I believe the issue here is that you imported `latLngBounds` directly from `"leaflet"` instead of from the ESM as described in the [readme](https://github.com/vue-leaflet/vue-leaflet/#working-with-leaflet), but also did not specify the `:use-global-leaflet` option as mentioned in the [issue linked from there](https://github.com/vue-leaflet/vue-leaflet/issues/48) about `latLngBounds` not working when imported from Leaflet, or [this discussion about window.L](https://github.com/vue-leaflet/vue-leaflet/discussions/161).

I'm ramping my activity back up here after being absent for a while. I think based on this issue, many others like it, and that discussion, one of the first things I'll try to get out after catching up on discussions and issues will be an update that works with standard Leaflet imports by default unless SSR compatibility is enabled, instead of vice versa.

As for padding your bounds, Leaflet can't perform the calculations linking geographic extents to pixels until after the map is created, so I don't believe there's any way to pre-pad it unless Leaflet itself were to provide such an option. The main approach I can think of would be to fit the bounds in the map's `@ready` handler, as described in [this comment](https://github.com/vue-leaflet/vue-leaflet/issues/48#issuecomment-733758917) and the one immediately after it.
